catanatron_core/catanatron/models/board.py

```diff
@@ -16,26 +16,39 @@
 )
 from catanatron.models.enums import FastBuildingType, SETTLEMENT, CITY
 
-
 # Used to find relationships between nodes and edges
 base_map = CatanMap.from_template(BASE_MAP_TEMPLATE)
 mini_map = CatanMap.from_template(MINI_MAP_TEMPLATE)
-STATIC_GRAPH = nx.Graph()
+STATIC_GRAPH_NX = nx.Graph()
 for tile in base_map.tiles.values():
-    STATIC_GRAPH.add_nodes_from(tile.nodes.values())
-    STATIC_GRAPH.add_edges_from(tile.edges.values())
+    STATIC_GRAPH_NX.add_nodes_from(tile.nodes.values())
+    STATIC_GRAPH_NX.add_edges_from(tile.edges.values())
```

Comment:
Do we still need this if we have the C++ graph? I would assume we only impose the cost of building _one_ of these. More to the point, I think it's best to abstract some of this away - maybe a subclass for the nx Graph with the corresponding methods forwarded to the C++ impl?

Reply:
At the moment, the C++ graph doesn't implement the methods necessary for the following two nx methods used by Catanatron: nx.floyd_warshall, nx.shortest_path. These methods depend on the internal implementation of  nx.Graph which differs in how the C++ version works.. so to replace, the C++ version would have to implement the corresponding methods, or Catantron would have to not use these features. 

It's certainly doable but it's a non-negligible amount of work, and the main gains can be achieved with what's implemented here. So I've opted for a hybrid version where catanatron_core defaults to the fast graph and when necessary, the rest of the code can fall back to the nx graph. The graphs are built once so there's no cost in having two of them other than code complexity. I agree it makes sense to structure them better. A subclass which routes the methods to the corresponding version sounds reasonable, I can look into that.

Reply:
yep, I agree! it's something I'm looking into and I'll integrate it into an updated version of this PR